evergreen.yml

```diff
@@ -335,47 +340,28 @@ functions:
       content_type: ${content_type|application/gzip}
       display_name: ${display_name}
 
-  ##
-  # Download one of genny's dependencies as a tarball and expand it
-  #
-  # Expansions:
-  #  install_dir: The directory to install this dependency
-  #  short_name: The name for the folder and prefix for the tarball to fetch
-  ###
-  f_fetch_library:
-  - command: s3.get
-    params:
-      aws_key: ${aws_key}
-      aws_secret: ${aws_secret}
-      extract_to: ${install_dir}
-      remote_file: ${project}/${short_name}/${short_name}-${build_variant}.tgz
-      bucket: mciuploads
-      permissions: public-read
-      content_type: ${content_type|application/gzip}
-
   ##
   # Compile the project in src/build.
-  #
-  # Expansions:
-  #  cmake: The CMake executable to use
-  #  toolchain: A toolchain directory to add to the PATH
-  ###
+  ##
   f_compile:
-  - command: subprocess.exec
+  - command: shell.exec
     params:
       working_dir: src
-      binary: /bin/bash
-      args:
-      - ${workdir}/src/scripts/bootstrap.sh
-      - "-DENABLE_JASPER:BOOL=ON"
-      - "-DGENNY_STATIC_BOOST_PATH:PATH=${workdir}/boost"
-      - "-DGENNY_INSTALL_DIR:PATH=${workdir}/dist"
-      - "-DGENNY_INSTALL_SSL:BOOL=ON"
-      env:
-        # This should be reasonably safe even on low memory machines
-        MAKEFLAGS: "-j8"
-        CMAKE: ${cmake|cmake}
-        PATH: ${toolchain}:/usr/bin:/bin
+      shell: bash
+      script: |
+        set -o errexit
+        set -o pipefail
+
+        # for python3
+        export PATH=/opt/mongodbtoolchain/v3/bin:$PATH
+
+        export DEVELOPER_DIR="/Applications/Xcode10.1.app"
+
+        ./scripts/lamp \
+          -d=${distro|not-linux} \
+          -DENABLE_JASPER:BOOL=ON \
+          -DGENNY_INSTALL_DIR:PATH=${workdir}/dist \
+          -DGENNY_INSTALL_SSL:BOOL=ON
```

Comment:
I think you need to quote each of these args to prevent spaces from wreaking havoc

```sh
./scripts/lamp \
  "-d=${distro|not-linux}" \
  ...
  "-DGENNY_INSTALL_DIR=${workdir/dist" \
```

don't technically need to quote the lines without var expansions but I would just for consistency